library/core/class.locale.php

```diff
@@ -357,7 +357,9 @@ public function translate($code, $default = false) {
 
         $this->EventArguments['Code'] = $code;
         $this->EventArguments['Default'] = $default;
-        $this->fireEvent('BeforeTranslate');
+        if (Gdn::config('Debug', false)) {
```

Comment:
Since this is so commonly called, my feeling is that this continuous lookup of this config value would be particularly expensive.

This should probably be use the static config cache trait you've written to prevent that.

I don't have a particular opinion about the key here, but Debug seems better than a brand new config key.

Reply:
I've tested - there is performance improvement when config() is checked. 
But in general I agree that config() call kind same "weight" of complexity.